app/components/sidebars/main/main_sidebar.js

```diff
@@ -181,6 +181,10 @@ export default class ChannelSidebar extends Component {
 
         this.closeChannelDrawer();
 
+        if (!channel) {
```

Comment:
Ok I see you are closing the drawer now, have you tried to actually identify why `channel` might be null or undefined here?

Reply:
I'd definitely like to know why undefined could be passed since I think this is only called by the channel list items when you tap on them. Alternatively, we could just add an alert modal with an error message instead of just closing the sidebar with no other feedback

Reply:
Keep this open for now. I will give this another shot.

Reply:
I did try. I again went through it. I cannot replicate this. I can add a modal with an error message.

Does `We couldn't open the channel you are trying to access. Please try again. If the problem persists please let us know` sound ok?.

Reply:
@esethna thoughts on the text message?

Reply:
@enahum @sudheerDev I don't understand what's happening to cause this. When would this message appear, what buttons would the user have on this modal and what would happen after the user taps those buttons? Why do we need such a message?

Reply:
Sorry, I should have made it clear. This happens when a user taps on a channel through LHS for joining a channel(i.e though jump to) and there is no channel information to join. There are few existing error handlers for network error and even deleted channel should not crash app.

Existing network error messages are
`We couldn't open a direct message with {displayName}. Please check your connection and try again.`

`We couldn't join the channel {displayName}. Please check your connection and try again.`

I cannot replicate how the user would reach a state as this. This PR prevents the app from crashing, closes the channel drawer and alerts the user with a message

Reply:
Chatted with Sudheer. Given that "try again" likely won't solve the issue, we think the best approach is going to be a modal similar to the cases above ^. With the text ``We couldn't join the channel. Please reset the cache and try again.``

Reply:
Made the change and this is how it looks.

<img width="366" alt="screen shot 2018-08-10 at 5 28 37 pm" src="https://user-images.githubusercontent.com/4973621/43956812-2dff05c4-9cc3-11e8-86bd-9861c07d455c.png">